File: removeelement.java
public class Solution {
    public int removeElement(int[] A, int elem) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int last = A.length-1;
        int i=0;
        while(i<=last){
            if(A[i]==elem){
                A[i]=A[last];
                last--;
            }
            else{
                i++;
            }
        }
        return i;
    }
}

File: cpp/RemoveElement.cpp
class Solution {
public:
    int removeElement(int A[], int n, int elem) {
        int right = n-1;
        int left = 0;
        while(left<=right) {
            if(A[left] == elem) {
                A[left] = A[right];
                right--;
            }
            else {
                left++;
            }
        }
        return right+1;
    }
};
